### ISSUE TITLE
[Question] Using Regula as a library

### ISSUE THREAD

**Issue by darrendao** [fugue/regula#254]
First off, thank you for open sourcing this wonderful project. Now to my question: what is Regula's stand on us using it as a library instead of a CLI?  To give more context, we're building a service for scanning IaC code. Basically doing what Regula does, but more service-oriented and having the policies/rules evaluation happened centrally rather than by each individual client/user. I don't see any API documentation for Regula but was able to look through the code under `pkg` and was able to do what I need to do. However, I want to know if aligns with Regula's vision. I don't want to go down this path if Regula decides later on to not support such usage anymore

**Comment by fugue-chris** [comment]
Hi Darren, thanks for reaching out. This is definitely a use case that we
allow and want to support going forward. Any chance you'd be willing to hop
on a call with a few of our engineers so that we can better understand your
needs?

**Comment by darrendao** [comment]
@fugue-chris Yes. Let's chat. Sorry late response. Was enjoying the long Thanksgiving break.

**Comment by fugue-chris** [comment]
@darrendao hope you had a restful and food-filled break! Do you mind sending an email to regula@fugue.co? We'll coordinate with you on a time.

**Comment by blancettb** [comment]
Hi guys,  could you share the outcome of this meeting? We are also interested in investigating using Regula in a service. We can brute force cli calls from a wrapped service but would rather use an official API.  Thanks for the project - it looks really nice.

**Comment by darrendao** [comment]
@blancettb I met with the Regula folks and they confirmed that they do intend for others to be able to use Regula as a library. There was some WIP to make it easier to use Regula as a library but I don't know if there's a concrete plan/timeline for an official API. Maybe @chris-snyk can give us an update.
